src/components/Dashboard/Utilization/UtilizationItem.js

```diff
@@ -1,69 +1,107 @@
 import React from 'react';
 import PropTypes from 'prop-types';
+import bounds from 'react-bounds';
 
 import { Col, Row, SparklineChart } from 'patternfly-react';
 
+import { MEDIA_QUERY_SM, MEDIA_QUERY_EXCLUSIVE_DEVIATION } from '../../../utils';
+
 import { InlineLoading } from '../../Loading';
 import { NOT_AVAILABLE } from './strings';
 
-export const UtilizationItem = ({ id, title, data, maxY, unit, isLoading, LoadingComponent }) => {
-  const axis = {
-    y: {
-      show: true,
-      padding: {
-        top: 0,
-        bottom: 0,
+class _UtilizationItem extends React.PureComponent {
+  static bounds() {
+    return {
+      standard: {
+        minWidth: MEDIA_QUERY_SM,
       },
-      tick: {
-        count: 3,
+      narrow: {
+        maxWidth: MEDIA_QUERY_SM - MEDIA_QUERY_EXCLUSIVE_DEVIATION,
       },
-    },
-    x: {
-      show: false,
-    },
-  };
-  if (maxY) {
-    axis.y.max = maxY;
+    };
   }
 
-  let actual;
-  let chartData;
-  if (!isLoading && data) {
-    actual = Math.round(data[data.length - 1]);
-    chartData = {
-      columns: [[unit, ...data.map(val => val.toFixed(1))]],
-      unload: true,
-      type: 'area',
+  render() {
+    const { isBound, id, title, data, maxY, unit, isLoading, LoadingComponent } = this.props;
+
+    const axis = {
+      y: {
+        show: true,
+        padding: {
+          top: 0,
+          bottom: 0,
+        },
+        tick: {
+          count: 3,
+        },
+      },
+      x: {
+        show: false,
+      },
     };
-  }
+    if (maxY) {
+      axis.y.max = maxY;
+    }
 
-  return (
-    <div className="kubevirt-utilization__item">
-      <Row>
-        <Col lg={2} md={2} sm={2} xs={2}>
-          {title}
-        </Col>
-        <Col className="kubevirt-utilization__item-actual" lg={3} md={3} sm={3} xs={3}>
-          {isLoading || !data || `${actual} ${unit}`}
-        </Col>
-        <Col className="kubevirt-utilization__item-chart" lg={7} md={7} sm={7} xs={7}>
-          {isLoading && <LoadingComponent />}
-          {!isLoading && data && <SparklineChart id={id} data={chartData} axis={axis} unloadBeforeLoad />}
-          {!isLoading && !data && NOT_AVAILABLE}
-        </Col>
-      </Row>
-    </div>
-  );
-};
+    let actual;
+    let chart = NOT_AVAILABLE;
+    if (isLoading) {
+      chart = <LoadingComponent />;
+    } else if (data) {
+      const chartData = {
+        columns: [[unit, ...data.map(val => val.toFixed(1))]],
+        unload: true,
+        type: 'area',
+      };
+      actual = `${Math.round(data[data.length - 1])} ${unit}`;
+      chart = <SparklineChart id={id} data={chartData} axis={axis} unloadBeforeLoad />;
+    }
+
+    if (isBound('narrow')) {
```

Comment:
maybe it would be easier to maintain if we keep only one return statement and parametrize Column sizes and Chart part. For the first div with different className with can use `classNames` function

Reply:
the component structure is different, I use two `<Row>` components in the `narrow` case